### PR TITLE
feat(migrations): add migration documentation

### PR DESCRIPTION
**Pull request by gabrielhase**

### CODE REVIEW

**Comment by gabrielhase** [comment]
@marcbachmann @DaRaFF  please check for correctness.
@mthpvg I assume you never used migrations before. This would make you a perfect fit to check this for understandabilty :smile:

File: data-migrations/migrations.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  Show migration report
+```
+
+Each migration is a 2-step process:
+1. Executing the migration
+2. Accepting the migration
+
+The option "Creates and prepares a migration" allows you to start a new migration. You can either choose:
+- a simple version bump which bumps the design version in all your documents to a new design version without affecting the structure of the documents
+- or select a file in which you wrote a script to tell the system how to migrate your documents (the next section will show some examples for such scripts)
```
Comment (DaRaFF) [inline data-migrations/migrations.md]: or select a migration script

File: data-migrations/migrations.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+Once you created a migration, it will run for all your documents. In the case of a migration script it will also generate a report for you that lists possible errors that were encountered with your script. (For a version bump there is no report, since errors are impossible). You can view this report by selecting the option "Show migration report".
```
Comment (DaRaFF) [inline data-migrations/migrations.md]: > since errors are impossible

haha :smile:

File: data-migrations/migrations.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+If after reviewing the report you don't want the migration to be applied, just select the option "Cancels a migration" and the changes will be discarded.
+
+If you want the migration to be applied, select "Accepts a migration" which will update all necessary database records and alters your documents.
```
Comment (DaRaFF) [inline data-migrations/migrations.md]: I would prefer to see a small state diagram instead of text.

**Comment by DaRaFF** [comment]
All in all the descrition is good and correct.

**Comment by gabrielhase** [comment]
Merging this now. @mthpvg you can still give your review here, would help to get feedback from someone who doesn't know this yet.

**Pull request merged**